### ISSUE TITLE
Boilerplate Repos Blog Post

### ISSUE THREAD

**Issue by MattSurabian** [bocoup/rwd-game-boiler#27]
Write a blog post about effective git strategies to use when building something based of a boilerplate repository.

Cover:
 - Forking :fork_and_knife: 
 - Git remotes :satellite: 
 - Pulling the latest upstream changes :boat: 
 - Conflict resolution :facepunch:

**Comment by MattSurabian** [comment]
https://bocoup.com/?p=3636&preview=true

**Comment by bobholt** [comment]
### Nits

> By setting up the main boilerplate repository as it’s own remote

its

> That means your local repository doesn’t know what its newly added 

it's (or better broken out as "it has")

### Content:

1. When I think about this approach to using boilerplates, I'm mainly concerned about the issue of trust with the upstream developer: will they and how often are they going to make breaking changes? Following this path with an untrustworthy upstream can lead to even more yak-shaving.

2. This seems like a huge argument in favor of releasing boilerplates versioned into NPM with sensible defaults and overridable configs, much like [hjs-webpack](/henrikjoreteg/hjs-webpack). I'd rather test out an update to the foundation of my application as a versioned module than dealing with rebasing/merging.

**Comment by MattSurabian** [comment]
Thanks @bobholt, I think your second content point is right on. As I was writing the final section I started feeling the same way. There's a challenge there to the boilerplate devs to produce something that can be consumed in that way. I'm not entirely sure how to work this into the post but I wonder how feasible it would be to structure the games boiler in such a way?

On the first point I think I can find places to insert the trust issue.

**Comment by labond** [comment]
I got small things. I enjoyed reading this! It flows very logically and was easy to follow, even for me! ;)

Direct Clone
Using git to clone boilerplate repos is a great start, but it can sometimes be confusing how to proceed from here

Suggest "it can be difficult to know how to proceed"


Even if you know in the end this is where you’re going to end up I still advocate for configuring an upstream remote in the beginning. 

Suggest "Even if you know this is where you're going to end up, I still advocate for..."
